File: cesar_sanchez/week04/mtalab/mta.rb
require 'pry'

$lN = {
  :stops => ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th']
}
$lL = {
  :stops => ['8th', '6th', 'Union Square', '3rd', '1st']
}
$l6 = {
  :stops => ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
}

def trip (line, station_in, station_out)

  flow = line[:stops]

  index_in = flow.index station_in
  index_out = flow.index station_out

  if index_in > index_out
    flow.reverse!
    index_in = index_out
    index_out = flow.index station_in
  end

  journey = flow[index_in + 1 ..index_out]
  total_stations = journey.size

  puts "You must travel through the following stops on the N line: #{ journey.join(", ") }."
  return total_stations
end


def plan_trip (line_in, station_in, line_out, station_out)

  total_stations = 0
  if line_in == line_out
    total_stations += trip(line_in, station_in, station_out)
    puts "#{total_stations} stops in total."
  else
    total_stations += trip(line_in, station_in, "Union Square")
    puts "Change at Union Square"
    total_stations += trip(line_out, "Union Square", station_out)
    puts "#{total_stations} stops in total."
  end
end

plan_trip($lN, "34th", $l6, "Grand Central")
binding.pry

File: cesar_sanchez/week02/metro/js/mta.js
const lines = {
  lN : {
  name: 'Line N',
  stops : ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
  },

  lL : {
  name: 'Line L',
  stops : ['8th', '6th', 'Union Square', '3rd', '1st'],
  },
  l6 : {
  name: 'Line 6',
  stops : ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place']
  }
}

$(lines).each( function() {
  $(this).each( function(){
    
  })
})

const trip = function (line,stationIn,stationOut) {
  let flow = [];
  if (line === 'N'){
    flow = lN.stops;
  }
  if (line === 'L'){
    flow = lL.stops;
  }
  if (line === '6'){
    flow = l6.stops;
  }
  if ( flow.indexOf(stationIn) > flow.indexOf(stationOut) ) {
    flow = flow.reverse()
  }
  const getIn = flow.indexOf(stationIn);
  const getOut = flow.indexOf(stationOut);
  let commute = 1;
  let stations = `You must travel through the following stops on the ${ line } line:`;
  stations = stations + ' ' + flow[getIn + 1]
  for (let i = (getIn + 2); i < (getOut + 1); i++) {
    commute = commute + 1;
    stations = stations + ', ' + flow[i];
  }
  console.log (stations);
  return commute;
}

const planTrip = function (lineIn, stationIn, lineOut, stationOut) {
  let commute = 0;
  if (lineIn === lineOut){
    commute = trip (lineIn, stationIn, stationOut);
    console.log(`${ commute } stops in total.`)
  } else {
    commute = trip (lineIn, stationIn, 'Union Square');
    console.log(`Change at Union Square and take the ${ lineOut } line.`)
    commute = commute + trip (lineOut, 'Union Square', stationOut);
    console.log(`${ commute } stops in total.`)
  }
}
